src/Storages/StorageMaterializedView.cpp

```diff
@@ -165,14 +165,20 @@ void StorageMaterializedView::drop()
     if (!select_query.select_table_id.empty())
         DatabaseCatalog::instance().removeDependency(select_query.select_table_id, table_id);
 
+    dropInnerTable(true);
+}
+
+void StorageMaterializedView::dropInnerTable(bool no_delay)
+{
     if (has_inner_table && tryGetTargetTable())
-        executeDropQuery(ASTDropQuery::Kind::Drop, global_context, target_table_id);
+        executeDropQuery(ASTDropQuery::Kind::Drop, global_context, target_table_id, no_delay);
+    has_inner_table = false;
```

Comment:
Maybe add `drop_query->if_exists = true` instead? It also will be helpful if inner table was dropped, but mv wasn't for some reason.

Reply:
Nice idea, done